### PR TITLE
WIP: Send dispatch event to buildpacks/samples on lifecycle release

### LINKED ISSUE #801: Add automation to re-generate sample builders when lifecycle is published
### Description
This will ensure the sample builders are kept up-to-date with the lifecycle.

### Proposed solution
[Slack thread](https://buildpacks.slack.com/archives/CST4A3ECV/p1638807846018600?thread_ts=1638800094.018200&cid=CST4A3ECV)

> Can I propose that the [#implementation](https://buildpacks.slack.com/archives/CSFE49VGR) team add automation to have this done similar to how we do with pack releases?
See [here](https://github.com/buildpacks/samples/blob/aa79de59c4bae24e32f15fda467d02ae9cd94b01/.github/workflows/build-and-deploy.yml#L10-L12) and [here](https://github.com/buildpacks/pack/blob/main/.github/workflows/delivery-release-dispatch.yml).
Additionally, we’d need to update the `builder.toml`[ files](https://github.com/search?q=repo%3Abuildpacks%2Fsamples+filename%3Abuilder.toml&type=Code&ref=advsearch&l=&l=) to set an explicit version of the lifecycle instead of relying on pack’s default since they don’t typically correlate.

### PR DESCRIPTION
**Pull request by glokta1**
Adds workflow to send dispatch event to `buildpacks/samples` on lifecycle releases. This will be used as a trigger on the receiving end to rebuild the samples and update their `builder.toml` files with the latest lifecycle version.

Lifecycle version is sent in dispatch payload.

Resolves #801 

**Question:** Am I correct in thinking the activity type of `release` should be `released` and not `published`? I took it from the `post-release.yml` file but I presume we want to send the dispatch event only for (stable) releases, not pre-releases or draft releases?

### CODE REVIEW

File: .github/workflows/release-dispatch.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: release-dispatch
+
+on:
+  release:
+    types:
+      - published
```
Comment (natalieparellano) [inline .github/workflows/release-dispatch.yml:6]: Great catch! I do think we want this to be `released`

File: .github/workflows/release-dispatch.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+jobs:
+  send-release-dispatch:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Repository Dispatch
+        uses: peter-evans/repository-dispatch@v2
+        with:
+          token: ${{ secrets.PLATFORM_GITHUB_TOKEN }}
```
Comment (natalieparellano) [inline .github/workflows/release-dispatch.yml:15]:
```suggestion
          token: ${{ secrets.GITHUB_TOKEN }}
```

**Comment by natalieparellano** [comment]
Thanks for the PR @glokta1! I'll test this out on my fork...

File: .github/workflows/release-dispatch.yml (hunk 2 above)
Comment (natalieparellano) [reply .github/workflows/release-dispatch.yml:15]: Oof, I thought the GITHUB_TOKEN would have enough permissions, but maybe not. Investigating...

File: .github/workflows/release-dispatch.yml (hunk 1 above)
Comment (natalieparellano) [inline .github/workflows/release-dispatch.yml:6]:
```suggestion
      - released
```

**Review by natalieparellano — COMMENTED** [review]
@glokta1 thanks for this! I added a couple of suggestions.

**Review by natalieparellano — APPROVED** [review]

**Comment by natalieparellano** [comment]
Looks good to me! @jabrown85 WDYT?

**Review by jabrown85 — APPROVED** [review]

**Comment by natalieparellano** [comment]
@glokta1 apologies for letting this PR sit here - I think we were waiting for https://github.com/buildpacks/samples/pull/138 to react to the dispatch before merging them together. It looks like https://github.com/buildpacks/samples/pull/138 registers the event but doesn't actually update the builder.toml files or published images. Let us know if you intend on picking that back up, otherwise we might just close both PRs for now.

**Comment by glokta1** [comment]
Hey @natalieparellano, I should apologize for the lack of communication. I encountered some difficulties in setting up the test environment while writing the workflow and left it at the time. I will pick this back up in a 2-3 days' time :)

**Comment by natalieparellano** [comment]
No worries, thanks a lot for your effort @glokta1 ❤️

**Pull request closed without merging**